crates/bitwarden/src/secrets_manager/secrets/update.rs

```diff
@@ -24,14 +28,15 @@ pub(crate) async fn update_secret(
     client: &mut Client,
     input: &SecretPutRequest,
 ) -> Result<SecretResponse> {
-    let enc = client.get_encryption_settings()?;
-
-    let org_id = Some(input.organization_id);
+    let key = client
+        .get_encryption_settings()?
+        .get_key(&Some(input.organization_id))
+        .ok_or(Error::VaultLocked)?;
 
     let secret = Some(SecretUpdateRequestModel {
-        key: enc.encrypt(input.key.as_bytes(), &org_id)?.to_string(),
-        value: enc.encrypt(input.value.as_bytes(), &org_id)?.to_string(),
-        note: enc.encrypt(input.note.as_bytes(), &org_id)?.to_string(),
+        key: input.key.clone().encrypt_with_key(key)?.to_string(),
+        value: input.value.clone().encrypt_with_key(key)?.to_string(),
+        note: input.note.clone().encrypt_with_key(key)?.to_string(),
```

Comment:
This makes me doubt that having encryptable interface consuming the string was correct 😩. @MGibson1 

Would `to_owned()` be more appropriate than `clone`?

Reply:
There is `to_owned()` and `to_string()`, either would work too, yeah.

Ultimately if we want the `Encryptable` interface to consume the decrypted items, we need to change the APIs to take the requests by value and not by reference, which is a breaking change, otherwise at some point we're going to need to copy the values to have ownership of them.